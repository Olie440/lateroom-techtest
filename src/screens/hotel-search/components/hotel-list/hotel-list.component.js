import React from 'react';
import { connect } from 'react-redux';

// Ideally we would have an id for each hotel we could use as the key
export function HotelRow({ name, starRating, facilities }, index) {
    return (
        <div className="hotel-list__row" key={index}>
            <p className="hotel-list__title">{name}</p>
            <p>Rating: {starRating}</p>
            <p>Facilities: {facilities.join(', ')}</p>
        </div>
    );
}

export function HotelList({ hotels }) {
    if (hotels.state !== 'Success') {
        return null;
    }

    return (
        <div className="hotel-list">
            { hotels.data.map(HotelRow) }
        </div>
    );
}

export function mapStateToProps(state) {
    return {
        hotels: state.hotels
    }
}

export default connect(mapStateToProps)(HotelList);
