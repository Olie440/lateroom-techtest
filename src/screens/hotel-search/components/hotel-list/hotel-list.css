.hotel-list {
    background: #ffffff;
    border: 1px solid #cccccc;
    border-top: none;
}

.hotel-list__row {
    padding: 1em;
}

.hotel-list__row:nth-child(even) {
    background: #f0f0f0;
}

.hotel-list__title {
    font-size: 1.25rem;
    font-weight: bold;
    margin-bottom: 0.5rem;
}

.hotel-list__sub {
    font-size: 0.75rem;
    color: #777777;
    margin-bottom: 0.25rem;
}
