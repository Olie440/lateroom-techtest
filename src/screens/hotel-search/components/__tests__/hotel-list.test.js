import React from "react";
import { HotelList } from "../hotel-list.component";
import mockHotels from "../../../../../__mocks__/mock-hotels";
import { shallow } from "enzyme";

describe("HotelList", () => {
  let component, props;

  beforeEach(() => {
    props = {
      loadHotels: jest.fn(),
      loading: false,
      error: false,
      loaded: false,
      hotels: []
    };
    component = shallow(<HotelList {...props} />);
  });

  it("calls the loadHotels action when loading = false, error = false and loaded = false", () => {
    expect(props.loadHotels).toHaveBeenCalled();
  });

  it("renders hotels correctly", () => {
    component.setProps({
      loaded: true,
      hotels: mockHotels().map((hotel, index) => ({ ...hotel, id: index }))
    });
    expect(component).toMatchSnapshot();
  });

  it("renders the loading component when loading = true", () => {
    component.setProps({
      loading: true
    });

    expect(component).toMatchSnapshot();
  });

  it("renders the error component when error = true", () => {
    component.setProps({
      error: true
    });

    expect(component).toMatchSnapshot();
  });
});
