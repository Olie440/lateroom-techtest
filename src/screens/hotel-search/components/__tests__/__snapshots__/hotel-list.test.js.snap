// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList renders hotels correctly 1`] = `
<styled.div>
  <styled.div
    key="0"
  >
    <styled.p>
      Hotelone
    </styled.p>
    <styled.p>
      <strong>
        Rating: 
      </strong>
      5
      /5
    </styled.p>
    <styled.p>
      <strong>
        Facilities: 
      </strong>
      Car park, Pool
    </styled.p>
  </styled.div>
  <styled.div
    key="1"
  >
    <styled.p>
      Hoteltwo
    </styled.p>
    <styled.p>
      <strong>
        Rating: 
      </strong>
      3
      /5
    </styled.p>
    <styled.p>
      <strong>
        Facilities: 
      </strong>
      Car park, Gym
    </styled.p>
  </styled.div>
  <styled.div
    key="2"
  >
    <styled.p>
      Hotelthree
    </styled.p>
    <styled.p>
      <strong>
        Rating: 
      </strong>
      3
      /5
    </styled.p>
    <styled.p>
      <strong>
        Facilities: 
      </strong>
    </styled.p>
  </styled.div>
</styled.div>
`;

exports[`HotelList renders the error component when error = true 1`] = `
<styled.div>
  <Error
    message="Error loading Hotel Data, please refresh to try again."
  />
</styled.div>
`;

exports[`HotelList renders the loading component when loading = true 1`] = `
<styled.div>
  <Loading
    message="Loading Hotel Data."
  />
</styled.div>
`;
